java-checks/src/test/files/checks/NullPointerCheck.java

```diff
@@ -0,0 +1,133 @@
+package javax.annotation;
+
+@interface CheckForNull {
+}
+
+@interface Nullable {
+}
+
+class NullPointerTest {
+
+  public Object[] notnullableField;
+
+  public Object[] notnullableMethod() {
+    return null;
+  }
+
+  @CheckForNull
+  public Object[] checkForNullField;
+
+  @CheckForNull
+  public Object[] checkForNullMethod() {
+    return null;
+  }
+
+  @Nullable
+  public Object[] nullableField;
+
+  @Nullable
+  public Object[] nullableMethod() {
+    return null;
+  }
+
+  public void testNotnullable(Object[] parameter) {
+    int i;
+    Object o;
+
+    Object[] array1 = notnullableField;
+    i = array1.length; // No issue
+    o = array1[0]; // No issue
+    array1.hashCode(); // No issue
+
+    i = notnullableField.length; // No issue
+    o = notnullableField[0]; // No issue
+    notnullableField.hashCode(); // No issue
+
+    Object[] array2 = notnullableMethod();
+    i = array2.length; // No issue
+    o = array2[0]; // No issue
+    array2.hashCode(); // No issue
+
+    i = notnullableMethod().length; // No issue
+    o = notnullableMethod()[0]; // No issue
+    notnullableMethod().hashCode(); // No issue
+
+    parameter.hashCode();
+  }
+
+  public void testCheckNotNull(@CheckForNull Object parameter) {
+    int i;
+    Object o;
+
+    Object[] array1 = checkForNullField;
+    i = array1.length; // False negative
+    o = array1[0]; // False negative
+    array1.hashCode(); // False negative
+
+    i = checkForNullField.length;
```

Comment:
Can you mark with a //NonCompliant comment the lines where you expect an issue ?
At least for consistency with our other checks and that really ease the reading of your test cases.